### ISSUE TITLE
Focused Launch: Programmatically highlight steps

### ISSUE THREAD

**Issue by tjcafferkey** [Automattic/wp-calypso#48048]
The current logic to highlight the correct steps [depends on us knowing the order of those steps](https://github.com/Automattic/wp-calypso/blob/trunk/packages/launch/src/focused-launch/summary/index.tsx#L593) up front. Meaning if we change the order of the steps, the highlighting will break unless we update this too.

We should approach this programmatically so that the highlighting logic will not depending on us knowing the order of every step for it to work.

**Comment by razvanpapadopol** [comment]
@tjcafferkey should we close this one now that https://github.com/Automattic/wp-calypso/issues/48047 is updated with latest decision on this?

**Comment by tjcafferkey** [comment]
@razvanpapadopol this one is slightly different, perhaps I've worded it wrong. Basically the one you've referenced (https://github.com/Automattic/wp-calypso/issues/48047) covers the edge case of the cart items.

This issue is to look at refactoring the logic for the linear flow so that we can re-order the steps of the summary items without changing any logic around highlighting. 

E.g. currently we say "if domains is completed, highlight plans as active". But what if domains and plans swapped order? Currently we would have to amend that logic.

**Comment by razvanpapadopol** [comment]
> re-order the steps of the summary items

> But what if domains and plans swapped order?

Let's first confirm with @ollierozdarz if re-ordering non-disabled steps in the UI is a possible solution. 

Until now, the only rule we have is that previously purchased items (domain and plan) are displayed greyed out at the top.

**Comment by tjcafferkey** [comment]
Yeah that's fine that is why it's marked as "Normal" priority although should may be changed to "Low" as currently there are no plans (that I'm aware of) to change the order.

**Comment by ollierozdarz** [comment]
> Let's first confirm with @ollierozdarz if re-ordering non-disabled steps in the UI is a possible solution.

Yeah, there should be no issue with having either domains or plans reordered to avoid breakages. I discussed this with @tjcafferkey a little in our call before Xmas to suit the scenario where a user will come from the cart. As long as the Title stays at the top (but still only show it when it's needed, and hide it when it's not) I think it's fine 🥇

**Comment by razvanpapadopol** [comment]
The issue is still valid even if the project is on hold.
